pkg/restic/backupper.go

```diff
@@ -184,7 +185,19 @@ ForEachVolume:
 	delete(b.results, resultsKey(pod.Namespace, pod.Name))
 	b.resultsLock.Unlock()
 
-	return volumeSnapshots, errs
+	for volumeName, snapshotID := range volumeSnapshots {
```

Comment:
It doesn't look like this approach is going to work - we need to capture the full PodVolumeBackup objects as they exist in the API server.  I would suggest changing the `volumeSnapshots` field to be a  `map[string]*velerov1api.PodVolumeBackup` or something similar, and then around https://github.com/heptio/velero/pull/1577/files#diff-d86635f0e9c2aff5dfda0f868fd8062fR170 you'll need to capture the PodVolumeBackups as they're completed/failed. Let me know if you need more input here.